server/src/main/java/org/elasticsearch/search/aggregations/bucket/composite/CompositeAggregationBuilder.java

```diff
@@ -61,6 +62,14 @@
         PARSER.declareObject(CompositeAggregationBuilder::aggregateAfter, (p, context) -> p.map(), AFTER_FIELD_NAME);
     }
 
+    public static void registerAggregators(ValuesSourceRegistry.Builder builder) {
+        DateHistogramValuesSourceBuilder.register(builder);
+        HistogramValuesSourceBuilder.register(builder);
+        GeoTileGridValuesSourceBuilder.register(builder);
+        TermsValuesSourceBuilder.register(builder);
+        builder.registerUsage(NAME);
```

Comment:
This preserves the existing behavior (i.e. it tracks composite usage like it was tracked before setting up the registry). It might be worth more detailed tracking, since we can capture what key types and values sources are being used.   But this PR is big enough already, and we don't have a clear plan for composite usage tracking, so I'm leaving it be for now.  Can open a ticket if folks think it's worth tinkering with, or we can just leave it alone.